### ISSUE TITLE
UART write does not work on QT Py SAMD21 with CircuitPython 8.x

### ISSUE THREAD

**Issue by alanmitchell** [adafruit/circuitpython#7612]
### CircuitPython version

```python
Tested with CP 8.0.2 and CP 20230218-12d3640.

Code works properly with CP 7.3.3.
```

### Code/REPL

```python
import time
import board
import busio

e5_uart = busio.UART(
    board.TX, board.RX,
    baudrate=9600,
    timeout=0.01
)

while True:
    out_bytes = bytes('From QT Py\n\r', 'utf-8')
    e5_uart.write(out_bytes)
    in_bytes = e5_uart.read(10)
    if in_bytes:
        print(in_bytes)

    time.sleep(1)
```

### Behavior

I have the QT Py Tx and Rx pins connected to a USB-TTL serial converter connected to my Linux PC.  For CP 8.x versions, no characters are received on the PC.  When I send characters from the PC to the USB-TTL converter, the QT Py does properly receive them, as indicated by characters being printed on the QT Py Serial console.

The above code works properly with CP 7.3.3.

**Comment by dhalbert** [comment]
Reproduced. Thanks -- I am taking a look.

**Comment by dhalbert** [comment]
Fixed by #7616.
